service/apigateway/marshal.go

```diff
@@ -0,0 +1,71 @@
+package apigateway
+
+import (
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/awserr"
+	"github.com/aws/aws-sdk-go-v2/aws/protocol/rest"
+	"github.com/aws/aws-sdk-go-v2/private/protocol/jsonrpc"
+	restlegacy "github.com/aws/aws-sdk-go-v2/private/protocol/rest"
+)
+
+// protoGetApiKeyMarshaler defines a marshaler for GetApiKey operation
+type protoGetApiKeyMarshaler struct {
+	input *GetApiKeyInput
+}
+
+// marshalOperation is the top level method used within a handler stack to marshal an operation
+// This method calls appropriate marshal shape functions as per the input shape and protocol used by the service.
+func (m protoGetApiKeyMarshaler) marshalOperation(r *aws.Request) {
+	var err error
+	encoder := rest.NewEncoder(r.HTTPRequest)
+	// adds content-type header
+	encoder.AddHeader("Content-Type").String("application/json")
```

Comment:
For the purpose of this prototype it would be helpful to call out what drives this header and when it should be set.